client/src/main/scala/drt/client/components/FlightComponents.scala

```diff
@@ -26,14 +23,27 @@ object FlightComponents {
     if (flightWithSplits.apiFlight.Origin.isDomesticOrCta)
       "pax-no-splits"
     else flightWithSplits.bestSplits match {
-      case Some(Splits(_, SplitSources.ApiSplitsWithHistoricalEGateAndFTPercentages, _, _)) => "pax-api"
+      case Some(Splits(_, SplitSources.ApiSplitsWithHistoricalEGateAndFTPercentages, _, _)) if isLiveAndApiPaxCountWithinThreshold(flightWithSplits) => "pax-api"
       case Some(Splits(_, SplitSources.PredictedSplitsWithHistoricalEGateAndFTPercentages, _, _)) => "pax-predicted"
       case Some(Splits(_, SplitSources.Historical, _, _)) => "pax-portfeed"
       case _ => "pax-unknown"
     }
 
+
+  def isLiveAndApiPaxCountWithinThreshold(flightWithSplits: ApiFlightWithSplits, threshold: Double = 0.05) : Boolean = {
+    if (flightWithSplits.apiFlight.FeedSources.contains(LiveFeedSource)) {
+    val apiSplits = flightWithSplits.splits.find(s => s.source == SplitSources.ApiSplitsWithHistoricalEGateAndFTPercentages)
+    val paxCount: Double = apiSplits.map(_.splits.toList.map(_.paxCount).sum).getOrElse(0)
+    val portDirectPax: Int = flightWithSplits.apiFlight.ActPax.getOrElse(0) - flightWithSplits.apiFlight.TranPax.getOrElse(0)
+    val isValidThreshold = paxCount != 0 && Math.abs(paxCount - portDirectPax) / paxCount < threshold
+      if (isValidThreshold) true else false
+    } else {
+      true
+    }
+  }
+
   def hasApiSplits(flightWithSplits: ApiFlightWithSplits): Boolean = flightWithSplits.bestSplits match {
```

Comment:
this function can then be replaced by a call to the one above probably.

Reply:
I am thinking to keep is separate function . As I think ,bestSplits is not needed to call for threshold check all time .

Reply:
I think the logic of `hasApiSplits` is now identical to your function above though right? It will always return the same result. Seems like duplication to me.